### DISCUSSION TITLE
Rfraw status reset

### DISCUSSION THREAD

**Discussion by bkbartk** [arendst/Tasmota#11447] [Q&A] [answered]
Hello,

For my sonoff RF bridge I use the latest version portisch firmware.
but sometimes my rfraw status get's reset automatically

> 16:02:33.514 MQT: tele/tasmota_EDEBDA/RESULT = {"Time":"2021-03-23T16:02:33","RfRaw":{"Data":"AA B1 03 0186 0852 0726 281818 55"}}
> 16:02:42.854 MQT: tele/tasmota_EDEBDA/RESULT = {"Time":"2021-03-23T16:02:42","RfRaw":{"Data":"AA B1 03 0208 03A2 0406 281808 55"}}
> 16:03:22.404 MQT: tele/tasmota_EDEBDA/STATE = {"Time":"2021-03-23T16:03:22","Uptime":"0T18:05:12","UptimeSec":65112,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"SomeSSID","BSSId":"xxxxx","Channel":9,"RSSI":80,"Signal":-60,"LinkCount":1,"Downtime":"0T00:00:05"}}
> 16:08:22.378 MQT: tele/tasmota_EDEBDA/STATE = {"Time":"2021-03-23T16:08:22","Uptime":"0T18:10:12","UptimeSec":65412,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"SomeSSID","BSSId":"xxxxx","Channel":9,"RSSI":82,"Signal":-59,"LinkCount":1,"Downtime":"0T00:00:05"}}
> 16:13:22.375 MQT: tele/tasmota_EDEBDA/STATE = {"Time":"2021-03-23T16:13:22","Uptime":"0T18:15:12","UptimeSec":65712,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"SomeSSID","BSSId":"xxxxx","Channel":9,"RSSI":78,"Signal":-61,"LinkCount":1,"Downtime":"0T00:00:05"}}
> 16:18:22.398 MQT: tele/tasmota_EDEBDA/STATE = {"Time":"2021-03-23T16:18:22","Uptime":"0T18:20:12","UptimeSec":66012,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"SomeSSID","BSSId":"xxxxx","Channel":9,"RSSI":78,"Signal":-61,"LinkCount":1,"Downtime":"0T00:00:05"}}
> 16:23:22.405 MQT: tele/tasmota_EDEBDA/STATE = {"Time":"2021-03-23T16:23:22","Uptime":"0T18:25:12","UptimeSec":66312,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"SomeSSID","BSSId":"xxxxx","Channel":9,"RSSI":78,"Signal":-61,"LinkCount":1,"Downtime":"0T00:00:05"}}
> 16:28:22.421 MQT: tele/tasmota_EDEBDA/STATE = {"Time":"2021-03-23T16:28:22","Uptime":"0T18:30:12","UptimeSec":66612,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"SomeSSID","BSSId":"xxxxx","Channel":9,"RSSI":80,"Signal":-60,"LinkCount":1,"Downtime":"0T00:00:05"}}
> 16:33:22.380 MQT: tele/tasmota_EDEBDA/STATE = {"Time":"2021-03-23T16:33:22","Uptime":"0T18:35:12","UptimeSec":66912,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"SomeSSID","BSSId":"xxxxx","Channel":9,"RSSI":72,"Signal":-64,"LinkCount":1,"Downtime":"0T00:00:05"}}
> 16:38:22.413 MQT: tele/tasmota_EDEBDA/STATE = {"Time":"2021-03-23T16:38:22","Uptime":"0T18:40:12","UptimeSec":67212,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"SomeSSID","BSSId":"xxxxx","Channel":9,"RSSI":82,"Signal":-59,"LinkCount":1,"Downtime":"0T00:00:05"}}
> 16:43:22.423 MQT: tele/tasmota_EDEBDA/STATE = {"Time":"2021-03-23T16:43:22","Uptime":"0T18:45:12","UptimeSec":67512,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"SomeSSID","BSSId":"xxxxx","Channel":9,"RSSI":80,"Signal":-60,"LinkCount":1,"Downtime":"0T00:00:05"}}
> 16:48:22.393 MQT: tele/tasmota_EDEBDA/STATE = {"Time":"2021-03-23T16:48:22","Uptime":"0T18:50:12","UptimeSec":67812,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"SomeSSID","BSSId":"xxxxx","Channel":9,"RSSI":72,"Signal":-64,"LinkCount":1,"Downtime":"0T00:00:05"}}
> 16:53:22.403 MQT: tele/tasmota_EDEBDA/STATE = {"Time":"2021-03-23T16:53:22","Uptime":"0T18:55:12","UptimeSec":68112,"Heap":27,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"SomeSSID","BSSId":"xxxxx","Channel":9,"RSSI":82,"Signal":-59,"LinkCount":1,"Downtime":"0T00:00:05"}}
> 16:57:07.334 CMD: rfraw
> 16:57:07.342 MQT: stat/tasmota_EDEBDA/RESULT = {"RfRaw":"ON"}

when I check it's set to on. but as you can see I don't retreive any data at all.
to fix This Issue, I added this to tasmota console

> Rule1 on system#boot do RfRaw AAB155 endon
> Rule1 1

and I added this automation to HA to run every 5 minutes

> - id: 'send RFRAW'
>   alias: send RFRAW
>   description: 'send RFRAW'
>   trigger:
>   - platform: time_pattern
>     minutes: '/5'
>   action:
>   - service: mqtt.publish
>     data:
>       topic: "cmnd/tasmota_EDEBDA/RfRaw"
>       payload: 'AAB155' 
>   - delay: 10
>   - service: mqtt.publish
>     data:
>       topic: "cmnd/tasmota_EDEBDA/RfRaw"
>       payload: 'AAB155'
>   mode: single

  andalso Run 'RFRAW AAB155' after every lightswitch command.
  this seems to work.
  But I think there should be an easier way to fix this which I am just missing.
  Can someone tell me how to retain RFRAW AAB155?
  AAB155 seems to work better then 177 an 1. however, If I use those I have the same issue

**Comment by Jason2866** [comment]
What version is latest Portisch firmware? There are version which are not running stable with rfraw

**Reply by bkbartk** [reply]
I got the one from tasmota repository
RF-Bridge-EFM8BB1-20190220.hex

Edit:
not sure which version should be ok or not. Any advise on this would be great
but today I was experiencing still some issues.
so I thought, based on your comment, let's give this one a try
RF-Bridge-EFM8BB1-20181127.hex
I see hex signals are totally different. but if it's more stable it's worth, changing. But I still need to test what it does in a couple of hours.

Edit2: this one totally crashes when you supply an invalid B0 code like this.
> RFRAW AA B0 0X 03 08 0208 03CA 07A8 01010102 55
at that point I need to unplug the power to have it working again

**Comment by bkbartk** [comment, marked as answer]
I now downloaded this version
https://github.com/Portisch/RF-Bridge-EFM8BB1/releases/tag/0x04
it's a little more consistent as far as I notice now, still some advice one which version I should use will be help full

And I came across this thread
https://github.com/Portisch/RF-Bridge-EFM8BB1/issues/175,
so next try

**Comment by bkbartk** [comment]
Ok,
final solution I hope,
I downloaded the source, included a couple of fixes made by halfbakery, not all because some didn't work for me.
I also added a few other fixes and created pushed them to github
https://github.com/bkbartk/RF-Bridge-EFM8BB1
for me those changes seem to work. But I don't know how to debug this firmware so there might be other issues which I cannot oversee.
The issue seems to be that the send code was too long for the buffer. it failed somewhere by splitting it up.
